relay_test.go

```diff
@@ -1496,85 +1498,174 @@ func inspectFrames(rh *relaytest.StubRelayHost) chan relay.CallFrame {
-				cf.Arg2Append(nil, nil)
-			},
-			modifyArg2: func(m map[string]string) map[string]string {
-				m[""] = ""
-				return m
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return addFixedKeyVal([]keyVal{{key: "", val: ""}})
 			},
 		},
 		{
 			msg: "add multiple zero-length key/value",
-			modifyFrame: func(cf relay.CallFrame, _ *relay.Conn) {
-				cf.Arg2Append(nil, nil)
-				cf.Arg2Append(nil, nil)
-				cf.Arg2Append(nil, nil)
-			},
-			modifyArg2: func(m map[string]string) map[string]string {
-				m[""] = ""
-				return m
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return addFixedKeyVal([]keyVal{
+					{"", ""},
+					{"", ""},
+					{"", ""},
+				})
 			},
 		},
 		{
 			msg: "add small key/value",
-			modifyFrame: func(cf relay.CallFrame, _ *relay.Conn) {
-				cf.Arg2Append([]byte("foo"), []byte("bar"))
-				cf.Arg2Append([]byte("baz"), []byte("qux"))
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return addFixedKeyVal([]keyVal{
+					{"foo", "bar"},
+					{"baz", "qux"},
+				})
 			},
-			modifyArg2: func(m map[string]string) map[string]string {
-				m["foo"] = "bar"
-				m["baz"] = "qux"
-				return m
+		},
+		{
+			msg: "fill the first frame until 2 bytes remain",
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return fillFrameWithArg2(t, cst, arg1, arg2, -2)
 			},
 		},
 		{
-			msg: "add large key/value",
-			modifyFrame: func(cf relay.CallFrame, _ *relay.Conn) {
-				cf.Arg2Append([]byte("fee"), []byte(largeVal1))
-				cf.Arg2Append([]byte("fi"), []byte(largeVal2))
-				cf.Arg2Append([]byte("fo"), []byte(largeVal3))
-				cf.Arg2Append([]byte("fum"), []byte(largeVal4))
+			msg: "fill the first frame until 1 byte remain",
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return fillFrameWithArg2(t, cst, arg1, arg2, -1)
+			},
+		},
+		{
+			msg: "fill the first frame to its boundary",
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return fillFrameWithArg2(t, cst, arg1, arg2, 0)
 			},
-			modifyArg2: func(m map[string]string) map[string]string {
-				m["fee"] = largeVal1
-				m["fi"] = largeVal2
-				m["fo"] = largeVal3
-				m["fum"] = largeVal4
-				return m
+		},
+		{
+			msg: "fill the first frame to 1 byte over its boundary",
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return fillFrameWithArg2(t, cst, arg1, arg2, 1)
+			},
+		},
+		{
+			msg: "fill the first frame to 2 bytes over its boundary",
+			modifier: func(t *testing.T, cst ChecksumType, arg1 string, arg2 map[string]string) relayModifier {
+				return fillFrameWithArg2(t, cst, arg1, arg2, 2)
+			},
+		},
+		{
+			msg: "add large key/value",
```

Comment:
can we expand this into 2 tests:
 - this one pushes the arg2 into a second frame
 - let's add another one which pushes arg2 into a 2nd and 3rd frame (e.g., we're adding closer to 128kb)